packages/rust-dprint-plugin-typescript/src/main.rs

```diff
@@ -0,0 +1,189 @@
+extern crate dprint_plugin_typescript as dprint;
+
+use clap::{App, Arg};
+use dprint::configuration::{Configuration, ConfigurationBuilder};
+use serde_json;
+use std::fs;
+use std::path::Path;
+use std::path::PathBuf;
+
+fn is_supported(path: &Path) -> bool {
+    if let Some(ext) = path.extension() {
+        if ext == "ts" || ext == "tsx" || ext == "js" || ext == "jsx" {
+            true
+        } else {
+            false
+        }
+    } else {
+        false
+    }
+}
+
+fn check_source_files(config: Configuration, paths: Vec<PathBuf>) -> Result<(), std::io::Error> {
+    let mut not_formatted_files = vec![];
+
+    for file_path in paths {
+        let file_path_str = file_path.to_string_lossy();
+        let file_contents = fs::read_to_string(&file_path).unwrap();
+        match dprint::format_text(&file_path_str, &file_contents, &config) {
```

Comment:
I really should just change this to just accept a `Path` object instead 🙂

I'll do that in the future.